tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js

```diff
@@ -0,0 +1,62 @@
+import { singularize } from 'ember-inflector';
+import { isPathParam, removePrefix, isSingleResourceIdentifier } from './resourceEvaluation.js';
+import { casing } from '@stoplight/spectral-functions';
+
+/**
+ * Returns IPA Compliant Operation ID.
+ *
+ * @param method the standard method name (create, update, get etc.), custom method name, or empty string (only for legacy custom methods)
+ * @param path the path for the endpoint
+ */
+export function generateOperationID(method, path) {
+  let resourceIdentifier = removePrefix(path);
+  let nouns = resourceIdentifier
+    .split('/')
+    .filter((id) => id.length > 0 && !isPathParam(id))
+    .map((noun) => capitalize(noun));
+
+  // legacy custom method - use end of path as custom method name
+  if (method === '') {
```

Comment:
```suggestion
  if (!method) {
```
In JS an empty string == undefined with evaluates as false in an if statement